src/NuGet.Core/NuGet.Build.Tasks/BuildTasksUtility.cs

```diff
@@ -695,6 +696,7 @@ public static string[] GetFallbackFolders(string projectDirectory, string[] fall
             return AppendItems(projectDirectory, currentFallbackFolders, filteredAdditionalProjectFallbackFolders);
         }
 
+        [Obsolete("This method calculates the source overrides incorrectly and will be removed in a future release")]
```

Comment:
The only difference I see between this `Obsolete` method and the new `GetSources` method implementation is addition of `startupDirectory` parameter and using its value to generate `sourcesOverride`. I would like to understand why new method was added instead of modifying the existing `GetSources()` method. The same comment is appliable to old `GetFallbackFolders` implementation that is marked as `Obsolete` in the PR.

Reply:
Public API: https://github.com/NuGet/NuGet.Client/pull/3722/files#diff-aeed904ce6c0c0d7930e3ba06196b571bcdba15ef45c6299a9111a1a78457054